### PR TITLE
[fix bug 1324722] Relocate Mozilla.Cookies helper to site.js bundle

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Also adds enabled() cookie feature detect function to helper

## Bugzilla link
https://bugzilla.mozilla.org/show_bug.cgi?id=1324722

## Testing
- Test to make sure no I didn't introduce any regressions, or miss any other pages/experiments?

## Checklist
- [ ] Requires l10n changes.
- [ ] Related functional & integration tests passing.

### CODE REVIEW

**Comment by alexgibson** [comment]
@jpetto r?

File: bedrock/settings/static_media.py [hunk 1]
```diff
@@ -1027,6 +1027,7 @@
             'js/newsletter/form.js',
             'js/base/mozilla-client.js',
             'js/base/mozilla-image-helper.js',
+            'js/base/mozilla-cookie-helper.js',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: As Traffic Cop runs before the `common` bundle is loaded, I think we need to add `mozilla-cookie-helper` to the `site` bundle.

File: docs/mozilla-traffic-cop.rst [hunk 2]
```diff
@@ -11,7 +11,8 @@ Mozilla Traffic Cop
 `mozilla-traffic-cop.js` handles redirecting users to different A/B/x variations
 through a query parameter.
 
-**`mozilla-traffic-cop.js` requires `mozilla-cookie-helper.js`.**
+**`mozilla-traffic-cop.js` requires `mozilla-cookie-helper.js`, which is included as
+default in the `common.js` bundle for most pages.**
```
Comment (jpetto) [inline docs/mozilla-traffic-cop.rst]: Update to `site.js` bundle?

File: media/js/base/mozilla-cookie-helper.js [hunk 3]
```diff
@@ -68,5 +68,32 @@ Mozilla.Cookies = {
...
+            // Create cookie
+            document.cookie = 'cookietest=1';
+            var ret = document.cookie.indexOf('cookietest=') !== -1;
+            // Delete cookie
+            document.cookie = 'cookietest=1; expires=Thu, 01-Jan-1970 00:00:01 GMT';
+            return ret;
+        }
+        catch (e) {
+            return false;
+        }
```
Comment (jpetto) [inline media/js/base/mozilla-cookie-helper.js]: Nice addition. 👍

File: media/js/base/mozilla-cookie-helper.js [hunk 4]
```diff
@@ -68,5 +68,32 @@ Mozilla.Cookies = {
         var aKeys = document.cookie.replace(/((?:^|\s*;)[^\=]+)(?=;|$)|^\s*|\s*(?:\=[^;]*)?(?:\1|$)/g, '').split(/\s*(?:\=[^;]*)?;\s*/);
         for (var nLen = aKeys.length, nIdx = 0; nIdx < nLen; nIdx++) { aKeys[nIdx] = decodeURIComponent(aKeys[nIdx]); }
         return aKeys;
+    },
+    enabled: function() {
+        /**
+         * Cookies deature detect lifted from Modernizr
```
Comment (craigcook) [inline media/js/base/mozilla-cookie-helper.js]: Typo: "feature"

File: media/js/base/mozilla-cookie-helper.js [hunk 5]
```diff
@@ -68,5 +68,32 @@ Mozilla.Cookies = {
...
+         * https://github.com/Modernizr/Modernizr/blob/master/feature-detects/cookies.js
+         *
+         * navigator.cookieEnabled cannot detect custom or nuanced cookie blocking
+         * configurations. For example, when blocking cookies via the Advanced
+         * Privacy Settings in IE9, it always returns true. And there have been
+         * issues in the past with site-specific exceptions.
+         * Don't rely on it.
+
+         * try..catch because some in situations `document.cookie` is exposed but throws a
```
Comment (craigcook) [inline media/js/base/mozilla-cookie-helper.js]: Typo: "because in some"

**Comment by alexgibson** [comment]
Thanks both - updated

**Comment by craigcook** [comment]
My sole contribution to this code review was proofreading comments. 🎓

**Pull request merged**